### PR TITLE
[WIP]Try to fix Serving installation issue

### LINKED ISSUE #829: Error happens occasionally when installing the nightly serving release
Not sure if it's an infra issue. Add it here now for record.

Detailed log: https://gubernator.knative.dev/build/knative-prow/pr-logs/pull/knative_test-infra/828/pull-knative-test-infra-integration-tests/1133108957661892608/
```
I0527 20:38:13.477] Installing Serving from https://storage.googleapis.com/knative-nightly/serving/latest/serving.yaml
I0527 20:38:13.937] namespace/knative-serving created
I0527 20:38:13.958] clusterrole.rbac.authorization.k8s.io/knative-serving-certmanager created
I0527 20:38:13.977] clusterrole.rbac.authorization.k8s.io/knative-serving-istio created
I0527 20:38:14.001] clusterrole.rbac.authorization.k8s.io/knative-serving-admin created
I0527 20:38:14.025] clusterrole.rbac.authorization.k8s.io/knative-serving-core created
I0527 20:38:14.082] serviceaccount/controller created
I0527 20:38:14.103] clusterrolebinding.rbac.authorization.k8s.io/knative-serving-controller-admin created
I0527 20:38:14.347] customresourcedefinition.apiextensions.k8s.io/certificates.networking.internal.knative.dev created
I0527 20:38:14.377] customresourcedefinition.apiextensions.k8s.io/clusteringresses.networking.internal.knative.dev created
I0527 20:38:14.407] customresourcedefinition.apiextensions.k8s.io/configurations.serving.knative.dev created
I0527 20:38:14.442] customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev created
I0527 20:38:14.480] customresourcedefinition.apiextensions.k8s.io/podautoscalers.autoscaling.internal.knative.dev created
I0527 20:38:14.516] customresourcedefinition.apiextensions.k8s.io/revisions.serving.knative.dev created
I0527 20:38:14.545] customresourcedefinition.apiextensions.k8s.io/routes.serving.knative.dev created
I0527 20:38:14.578] customresourcedefinition.apiextensions.k8s.io/services.serving.knative.dev created
I0527 20:38:14.615] customresourcedefinition.apiextensions.k8s.io/serverlessservices.networking.internal.knative.dev created
I0527 20:38:14.659] service/activator-service created
I0527 20:38:14.690] service/controller created
I0527 20:38:14.728] service/webhook created
I0527 20:38:14.771] deployment.apps/activator created
I0527 20:38:14.811] service/autoscaler created
I0527 20:38:14.840] deployment.apps/autoscaler created
I0527 20:38:14.874] configmap/config-autoscaler created
I0527 20:38:14.899] configmap/config-certmanager created
I0527 20:38:14.923] configmap/config-defaults created
I0527 20:38:14.949] configmap/config-deployment created
I0527 20:38:14.974] configmap/config-domain created
I0527 20:38:14.996] configmap/config-gc created
I0527 20:38:15.019] configmap/config-istio created
I0527 20:38:15.046] configmap/config-logging created
I0527 20:38:15.077] configmap/config-network created
I0527 20:38:15.105] configmap/config-observability created
I0527 20:38:15.127] configmap/config-tracing created
I0527 20:38:15.151] deployment.apps/controller created
I0527 20:38:15.177] deployment.apps/networking-certmanager created
I0527 20:38:15.209] deployment.apps/networking-istio created
I0527 20:38:15.238] deployment.apps/webhook created
I0527 20:38:15.238] unable to recognize "https://storage.googleapis.com/knative-nightly/serving/latest/serving.yaml": no matches for kind "Gateway" in version "networking.istio.io/v1alpha3"
I0527 20:38:15.239] unable to recognize "https://storage.googleapis.com/knative-nightly/serving/latest/serving.yaml": no matches for kind "Gateway" in version "networking.istio.io/v1alpha3"
I0527 20:38:15.239] unable to recognize "https://storage.googleapis.com/knative-nightly/serving/latest/serving.yaml": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
```

### PR DESCRIPTION
**Pull request by Fredy-Z**
Fixes #829

### CODE REVIEW

File: scripts/e2e-tests.sh [hunk 1]
```diff
@@ -211,6 +211,7 @@ function create_test_cluster() {
     ${extra_flags[@]} \
     ${EXTRA_KUBETEST_FLAGS[@]}
   echo "Test subprocess exited with code $?"
+  (( ! SKIP_ISTIO_ADDON )) && (wait_until_batch_job_complete istio-system || return 1)
```
Comment (adrcunha) [inline scripts/e2e-tests.sh]: Wrong place. This should be in setup_test_cluster().

Comment (Fredy-Z) [reply scripts/e2e-tests.sh]: The PR is WIP and I'm trying to find the root cause, but seems it's not.
From the log it's complaining the `Gateway` CRD of istio and `Image` CRD of knative caching are not found.

Comment (adrcunha) [reply scripts/e2e-tests.sh]: https://github.com/knative/test-infra/issues/829#issuecomment-496323288

**Pull request closed without merging**